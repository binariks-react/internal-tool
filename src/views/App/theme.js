export const getColor = (color, props) => (
  props ? props.theme.colors[color] : ({ theme }) => theme.colors[color]
);

export const getActiveColor = (color, props) => (
  props ? props.theme.colors.activeColors[color] : ({ theme }) => theme.colors.activeColors[color]
);

export const getHoverColor = (color, props) => (
  props ? props.theme.colors.hoverColors[color] : ({ theme }) => theme.colors.hoverColors[color]
);

export const getHoverShadowColor = (color, props) => (
  props ? props.theme.shadows.hover[color] : ({ theme }) => theme.shadows.hover[color]
);

export const getFocusShadowColor = (color, props) => (
  props ? props.theme.shadows.focus[color] : ({ theme }) => theme.shadows.focus[color]
);

export default {
  colors: {
    text: '#5a6169',
    lightGrey: '#e9ecef',
    darkGrey: '#becad6',
    border: '#e1e5eb',
    primary: '#007bff',
    secondary: '#5a6169',
    success: '#17c671',
    danger: '#c4183c',
    warning: '#ffb400',
    info: '#00b8d8',
    dark: '#212529',
    white: '#ffffff',
    royalBlue: '#674eec',
    java: '#1adba2',
    salmon: '#ff4169',
    hoverColors: {
      lightGrey: '#e9ecef',
      primary: '#006fe6',
      secondary: '#4e545b',
      success: '#14af64',
      danger: '#ad1535',
      warning: '#e6a200',
      info: '#00a2bf',
      dark: '#16181b',
      white: '#ffffff',
      royalBlue: '#5337ea',
      java: '#17c491',
      salmon: '#ff2855',
    },
    activeColors: {
      primary: '#0062cc',
      secondary: '#42484e',
      lightGrey: '#FBFBFB',
      success: '#129857',
      danger: '#97122e',
      warning: '#cc9000',
      info: '#008da5',
      dark: '#0a0c0d',
      white: '#007bff',
      royalBlue: '#3f20e7',
      java: '#15ad80',
      salmon: '#ff0e41',
    },
  },
  shadows: {
    hover: {
      primary: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(0,123,255,.25)',
      secondary: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(90,97,105,.25)',
      success: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(23,198,113,.25)',
      danger: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(196,24,60,.25)',
      warning: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(255,180,0,.25)',
      info: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(0,184,216,.25)',
      dark: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(33,37,41,.25)',
      white: '0 0.125rem 0.625rem rgba(129,142,163,.2), 0 0.0625rem 0.125rem rgba(129,142,164,.3)',
      royalBlue: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(103,78,236,.25)',
      java: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(26,219,162,.25)',
      salmon: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(255,65,105,.25)',
      inset: 'inset 0 3px 5px rgba(0,0,0,.125)!important',
    },
    focus: {
      primary: '0 0 0 3px rgba(0,123,255,.15),0 3px 15px rgba(0,123,255,.2),0 2px 5px rgba(0,0,0,.1)',
      secondary: '0 0 0 3px rgba(90,97,105,.15),0 3px 15px rgba(90,97,105,.2),0 2px 5px rgba(0,0,0,.1)',
      success: '0 0 0 3px rgba(23,198,113,.15),0 3px 15px rgba(23,198,113,.2),0 2px 5px rgba(0,0,0,.1)',
      danger: '0 0 0 3px rgba(196,24,60,.15),0 3px 15px rgba(196,24,60,.2),0 2px 5px rgba(0,0,0,.1)',
      warning: '0 0 0 3px rgba(255,180,0,.15),0 3px 15px rgba(255,180,0,.2),0 2px 5px rgba(0,0,0,.1)',
      info: '0 0 0 3px rgba(0,184,216,.15),0 3px 15px rgba(0,184,216,.2),0 2px 5px rgba(0,0,0,.1)',
      dark: '0 0 0 3px rgba(33,37,41,.15),0 3px 15px rgba(33,37,41,.2),0 2px 5px rgba(0,0,0,.1)',
      white: '0 0 0 3px rgba(251,251,251,.15),0 3px 15px rgba(251,251,251,.2),0 2px 5px rgba(0,0,0,.1)',
      royalBlue: '0 0 0 3px rgba(103,78,236,.15),0 3px 15px rgba(103,78,236,.2),0 2px 5px rgba(0,0,0,.1)',
      java: '0 0 0 3px rgba(26,219,162,.15),0 3px 15px rgba(26,219,162,.2),0 2px 5px rgba(0,0,0,.1)',
      salmon: '0 0 0 3px rgba(255,65,105,.15),0 3px 15px rgba(255,65,105,.2),0 2px 5px rgba(0,0,0,.1)',
      switch: '0 3px 8px 1px rgba(0,0,0,.2)',
    },
  },
};
