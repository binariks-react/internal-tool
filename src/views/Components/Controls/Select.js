import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Wrapper = styled.div`
  min-width: 7.5rem;
  width: 100%;
`;

const SelectWrapper = styled.select`
  width: 100%;
  padding: .4375rem .75rem;
  flex: 1 1 auto;
  font-size: .8125rem;
  line-height: 1.5;
  color: ${props => props.theme.colors.text};
  background-color: ${props => props.theme.colors.white};
  border: 1px solid ${props => props.theme.colors.border};
  font-weight: 300;
  border-radius: .25rem;
  box-shadow: none;
  transition: box-shadow 250ms cubic-bezier(.27,.01,.38,1.06), border 250ms cubic-bezier(.27,.01,.38,1.06);

  &:hover {
    border-color: ${props => props.theme.colors.hoverColors.border};
  }

  &:focus {
    color: ${props => props.theme.colors.text};
    background-color: ${props => props.theme.colors.white};
    border-color: ${props => props.theme.colors.activeColors.white};
    box-shadow: 0 0.313rem 0.719rem rgba(0,123,255,.1), 0 0.156rem 0.125rem rgba(0,0,0,.06);
  }
`;

const Select = ({ onChangeOption, ...props }) => {
  const changeOption = (e) => {
    const option = props.options.find(op => parseInt(op.value) === parseInt(e.target.value));
    onChangeOption && onChangeOption(option);
  };

  return (
    <Wrapper>
      <SelectWrapper {...props} defaultValue={-1} onChange={changeOption}>
        <option value={-1} disabled hidden>{props.placeholder}</option>
        {props.options && props.options.length > 0 && props.options.map(option => (
          <option key={option.value} value={option.value}>{option.label}</option>
        ))}
      </SelectWrapper>
    </Wrapper>);
};

Select.defaultProps = {
  placeholder: 'Select option',
  onChangeOption: () => {},
};

Select.propTypes = {
  options: PropTypes.arrayOf(PropTypes.shape({
    value: PropTypes.number.isRequired,
    label: PropTypes.string.isRequired,
  })).isRequired,
  placeholder: PropTypes.string,
  onChangeOption: PropTypes.func,
};

export default Select;
