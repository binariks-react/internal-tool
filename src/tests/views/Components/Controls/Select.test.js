import React from 'react';
import { ThemeProvider } from 'styled-components';
import { render, mount } from 'enzyme';
import theme from 'views/App/theme';
import Select from 'views/Components/Controls/Select';
import 'jest-styled-components';

describe('Select', () => {
  const fakeOptions = [
    {
      value: 1,
      label: 'First',
    },
    {
      value: 2,
      label: 'Second',
    },
    {
      value: 3,
      label: 'Third',
    },
  ];
  describe('snapshot', () => {
    it('select with options', () => {
      const input = render(
        <ThemeProvider theme={theme}>
          <Select options={fakeOptions} />
        </ThemeProvider>
      );
      expect(input).toMatchSnapshot();
    });

    it('select input with options and onChangeOption', () => {
      const input = render(
        <ThemeProvider theme={theme}>
          <Select onChangeOption={(option) => {}} options={fakeOptions} />
        </ThemeProvider>
      );
      expect(input).toMatchSnapshot();
    });
  });

  it('onChangeOption', () => {
    const onChangeOption = jest.fn();

    const select = mount(
      <ThemeProvider theme={theme}>
        <Select onChangeOption={onChangeOption} options={fakeOptions} />
      </ThemeProvider>
    );

    select.find('select').simulate('change', { target: fakeOptions[1] });
    expect(onChangeOption).toHaveBeenCalledTimes(1);
    expect(onChangeOption).toHaveBeenCalledWith(fakeOptions[1]);
  });

  it('onChangeOption - defaultProps', () => {
    expect(Select.defaultProps.onChangeOption).toBeDefined();
    expect(Select.defaultProps.onChangeOption()).toBeUndefined();
  });
});
