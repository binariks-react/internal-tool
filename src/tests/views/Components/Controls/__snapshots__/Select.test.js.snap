// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select snapshot select input 1`] = `
.c0 {
  min-width: 7.5rem;
  width: 100%;
}

.c1 {
  width: 100%;
  padding: .4375rem .75rem;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  font-size: .8125rem;
  line-height: 1.5;
  color: #5a6169;
  background-color: #ffffff;
  border: 1px solid #e1e5eb;
  font-weight: 300;
  border-radius: .25rem;
  box-shadow: none;
  -webkit-transition: box-shadow 250ms cubic-bezier(.27,.01,.38,1.06),border 250ms cubic-bezier(.27,.01,.38,1.06);
  transition: box-shadow 250ms cubic-bezier(.27,.01,.38,1.06),border 250ms cubic-bezier(.27,.01,.38,1.06);
}

.c1:hover {
  border-color: #b3bdcc;
}

.c1:focus {
  color: #5a6169;
  background-color: #ffffff;
  border-color: #007bff;
  box-shadow: 0 0.313rem 0.719rem rgba(0,123,255,.1),0 0.156rem 0.125rem rgba(0,0,0,.06);
}

<div
  class="c0"
>
  <select
    class="c1"
    placeholder="Select option"
  >
    <option
      disabled=""
      hidden=""
      selected=""
      value="-1"
    >
      Select option
    </option>
    <option
      value="1"
    >
      First
    </option>
    <option
      value="2"
    >
      Second
    </option>
    <option
      value="3"
    >
      Third
    </option>
  </select>
</div>
`;
